agones/src/xds.rs

```diff
@@ -181,7 +182,7 @@ filters:
             .as_mut()
             .map(|annotations| annotations.remove(token_key).unwrap());
         gameservers.replace(name.as_str(), &pp, &gs).await.unwrap();
-
+        debug_pods(client, "role=proxy");
```

Comment:
My suggestion might be to change line 201 from

```rust
assert!(failed, "Packet should have failed");
```

to something like:

```rust
if failed {
  // proxies
  debug_pods(client, "role=proxy");
  // xds server
  debug_pods(client, "role=xds");

  panic!("Packet should have failed");
}
```

Then the condition can always stay in the test in case of future debugging requirements.

Reply:
SGTM